dsl_parser/tests/test_register_function.py

```diff
@@ -80,29 +77,29 @@ def validate(self, plan):
                                                          'attribute']}},
                          outputs['output3']['value'])
 
-        def get_node_instances(node_id=None):
-            return [
-                NodeInstance({
-                    'id': 'webserver1',
-                    'node_id': 'webserver',
-                    'runtime_properties': {
-                        'attribute': 'attribute_value'
-                    }
-                })
-            ]
-
-        def get_node_instance(node_instance_id):
-            return get_node_instances()[0]
-
-        def get_node(node_id):
-            return Node({'id': node_id})
-
-        o = functions.evaluate_outputs(parsed['outputs'],
-                                       get_node_instances,
-                                       get_node_instance,
-                                       get_node,
-                                       self._get_secret_mock,
-                                       self._get_capability_mock)
+        class Storage(object):
+            def get_node_instances(self, node_id=None):
+                return [
+                    NodeInstance({
+                        'id': 'webserver1',
+                        'node_id': 'webserver',
+                        'runtime_properties': {
+                            'attribute': 'attribute_value'
+                        }
+                    })
+                ]
+
+            def get_node_instance(self, node_instance_id):
+                return self.get_node_instances()[0]
+
+            def get_node(self, node_id):
+                return Node({'id': node_id})
+
+        storage = Storage()
+        storage.get_secret = self._get_secret_mock
+        storage.get_capability = self._get_capability_mock
+
+        o = functions.evaluate_outputs(parsed['outputs'], storage)
```

Comment:
let's change 'o' to something more meaningful

Reply:
refactored, however note that this wasn't added in this pr